### ISSUE TITLE
Missing / custom domains - new medium domains are being added regularly

### ISSUE THREAD

**Issue by rblinzler** [manojVivek/medium-unlimited#105]
Medium is using more and more 'custom' domains (i.e. name_of_author.medium.com, js.plainenglish.io, etc.). I discover new ones almost every week. There is no way this add-on can stay ahead of the curve as long as the domains are hard coded. 

IMHO, the most useful solution would be a feature that would allow the user to edit the url list in settings. Outside of that, the code package does not contain required node packages. A detailed set of instructions on how to add custom domains and build the project would be very helpful.

**Comment by y2kviv** [comment]
I agree with rablinz. it would be more convenient if the user can add the new custom domain URL manually in the settings or maybe the extension can suggest the user to add the custom domain to 'medium-unlimited' through a small popup on top right (Refer the attached example) when the user visits the particular locked custom domain.
<img width="384" alt="popup" src="https://user-images.githubusercontent.com/80037570/111603037-64054d00-87ed-11eb-9d87-c1a39f037926.png">
.

@manojVivek Would you consider adding this feature mentioned? Do you plan on working on this feature request?

**Comment by Aarekaz** [comment]
This is really needed. A lot of popular pages are using custom domains.

**Comment by Plutonium-239** [comment]
I mean.. it is pretty easy to add your custom domains. Go to the unpacked folder of the extension (focusing on chrome here), open manifest.json in any text editor, and add your website in both the 'lists' that are visible, in the same format as the other websites (JSON formatting). save it and load the extension again in chrome. It should work then.

**Comment by Aarekaz** [comment]
Will try this @Plutonium-239

**Comment by y2kviv** [comment]
> I mean.. it is pretty easy to add your custom domains. Go to the unpacked folder of the extension (focusing on chrome here), open manifest.json in any text editor, and add your website in both the 'lists' that are visible, in the same format as the other websites (JSON formatting). save it and load the extension again in chrome. It should work then.

@Plutonium-239  I have tried this exact method previously but no luck. After editing the manifest.json it only unlocks a small part of the entire article apparently. Is there any other way to get it working?

Can you try this link yourself and let me know if it works for you after editing the manifest.json?
https://bootcamp.uxdesign.cc/goodbye-8-point-grid-hello-4-point-grid-1aa7f2159051

Before editing manifest.json
<img width="1041" alt="before" src="https://user-images.githubusercontent.com/80037570/113348705-feb26f80-9347-11eb-8291-15f140426c13.png">

After editing manifest.json
<img width="1259" alt="after" src="https://user-images.githubusercontent.com/80037570/113348694-fa865200-9347-11eb-92ea-def34fd1cdf5.png">

**Comment by Plutonium-239** [comment]
Okay, @y2kviv @Aarekaz @rablinz I have found a way to add any custom sites(one extra step from what I wrote above). Complete steps:
1. Locate the medium-unlimited folder (the unpacked extension)
2. Open manifest.json and add the URL (in JSON formatting) like all the other websites to BOTH the lists (visible)
3. Save and close
4. Open bakground.bundle.js and go to the end, you will see another list (lots of sites)
![image](https://user-images.githubusercontent.com/29684689/113436194-e868ea00-9401-11eb-9acf-ace898ebfb6c.png)
Add your website here too.

I checked it by adding https://bootcamp.uxdesign.cc/ and it unlocked the whole article properly.

**Comment by Aarekaz** [comment]
@Plutonium-239 I followed your above steps( the one before) and It worked. The only thing left is the background.bundle.js file edit. I will add it there too.

**Comment by y2kviv** [comment]
@Plutonium-239 Thanks for the suggestion.
 Editing bakground.bundle.js unlocked the whole article, but somehow it managed to break a few elements on the page for me.

See the screenshot below for the comparison.

Before the last free member-only article of the month (3/3)
![before last free member article](https://user-images.githubusercontent.com/80037570/113478803-38948a80-949c-11eb-904f-2b8add3a7438.png)

After unlocking the article (you can see the Twitter widget, sidebar, author details, and 'more from medium' section are missing from the article.
<img width="1222" alt="After unlocking the article" src="https://user-images.githubusercontent.com/80037570/113478897-c53f4880-949c-11eb-81d2-3cba33416880.png">

Missing elements

Author Details & More From Medium
<img width="1249" alt="Author   More from medium" src="https://user-images.githubusercontent.com/80037570/113479003-66c69a00-949d-11eb-82af-53f755e0e679.png">

Sidebar
<img width="256" alt="Sidebar missing" src="https://user-images.githubusercontent.com/80037570/113479007-6cbc7b00-949d-11eb-86f0-dafe9367476f.png">

Twitter Widget
<img width="282" alt="Twitter widget" src="https://user-images.githubusercontent.com/80037570/113479026-86f65900-949d-11eb-8321-b6b11eb963e4.png">

Not sure if it is a bug or not. Do you have any idea what might be causing this?

**Comment by Aarekaz** [comment]
@y2kviv No idea. @Plutonium-239  Any info? I have not checked this.

**Comment by Plutonium-239** [comment]
Hmm, I didn't notice this 
I will try to fiddle around in the code (although I don't really know js) but if there's anything logical, I can try that

**Comment by Plutonium-239** [comment]
@y2kviv definitely a bug tho, since when normal articles(like towardsdatascience or medium) are unlocked they are... normal.

**Comment by y2kviv** [comment]
> @y2kviv definitely a bug tho, since when normal articles(like towardsdatascience or medium) are unlocked they are... normal.

I have noticed it is breaking the elements mostly on custom domains somehow.

**Comment by vietvudanh** [comment]
> I mean.. it is pretty easy to add your custom domains. Go to the unpacked folder of the extension (focusing on chrome here), open manifest.json in any text editor, and add your website in both the 'lists' that are visible, in the same format as the other websites (JSON formatting). save it and load the extension again in chrome. It should work then.

How about on Firefox? I have tried to edit the xpi file by uzip and zip it again, but when I install from file, Firefox throw error that the file is corrupted
